### PR TITLE
[RPA-844] Bug fix for bad substitution of objects and lists to # variable

### PR DESCRIPTION
**Pull request by marcin-depcik**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-common/src/Expressions/index.ts [hunk 1]
```diff
@@ -68,10 +68,28 @@ export class Expressions {
             };
 
             response.jexl = true;
-            response.result = templatedString.replace(jexlPattern, (expressionMatch, innerProperty) => {
-                const evaluatedProperty = Jexl.evalSync(innerProperty, jexlContext);
-                return evaluatedProperty;
-            })
+
+            let isPropertyArray = false;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-common/src/Expressions/index.ts]: avoid using let. function programming/useState only

Comment (star-in-the-jar) [reply runbotics/runbotics-common/src/Expressions/index.ts]: I think the simplest is replacing:
if (isPropertyArray) {

with:
if(Array.isArray(JSON.parse(property)))

File: runbotics/runbotics-common/src/Expressions/index.ts [hunk 2]
```diff
@@ -68,10 +68,28 @@ export class Expressions {
...
+                response.result = JSON.parse(property);
+            } else if (property === "true") {
+                response.result = true;
+            } else if (property === "false") {
+                response.result = false;
+            } else if (property === "null") {
+                response.result = null;
+            } else {
+                response.result = property;
+            }
```
Comment (star-in-the-jar) [inline runbotics/runbotics-common/src/Expressions/index.ts]: maybe use switch?

**Review by star-in-the-jar — CHANGES_REQUESTED** [review]

**Review by preludium — DISMISSED** [review]

File: runbotics/runbotics-common/src/Expressions/index.ts (hunk 1 above)
Comment (marcin-depcik) [reply runbotics/runbotics-common/src/Expressions/index.ts]: In this case I need information about evaluatedProperty specifically from inside of replace function. Without JSON.stringify, replace will return an array parsed to String, which causes the bug.

**Review by star-in-the-jar — DISMISSED** [review]

**Pull request merged**